src/spyglass/spikesorting/utils_burst.py

```diff
@@ -0,0 +1,264 @@
+
+    for pair in pairs:
+        kwargs = (
+            dict(fig=None, axes=None, row_duration=100) if overlap else dict()
+        )
+
+        for u in pair:
+            ret1, ret2 = plot_burst_one_peak(
+                peak_v[u], peak_t[u], show_plot=overlap, **kwargs
+            )
+            if overlap:
+                fig, axes = ret1, ret2
+                kwargs = dict(fig=fig, axes=axes)
+            else:
+                if fig is None:
+                    fig, axes = dict(), dict()
+                fig[u], axes[u] = ret1, ret2
+
+        axes[0, 0].set_title(f"pair:{pair}")
+
+    return fig
+
+
+def plot_burst_by_sort_group(fig):
+    for sg_id, f in fig.items():
+        managed_fig, _ = plt.subplots(
+            1, 4, figsize=(24, 4), sharex=False, sharey=False
+        )
+        canvas_manager = managed_fig.canvas.manager
+        canvas_manager.canvas.figure = f
+        managed_fig.set_canvas(canvas_manager.canvas)
+        plt.suptitle(f"sort group {sg_id}", fontsize=20)
+    return fig
+
+
+def plot_burst_metrics(
+    sg_query: Union[List[Dict], QueryExpression],
+) -> plt.Figure:
+    """Parameters are 4 metrics to be plotted against each other.
+
+    Parameters
+    ----------
+    sg_query : Union[List[Dict], QueryExpression]
+        Query result or list of dictionaries with the following keys:
+        wf_similarity : dict
+            waveform similarities
+        isi_violation : dict
+            isi violation
+        xcorrel_asymm : dict
+            spike cross correlogram asymmetry
+
+    Returns
+    -------
+    figure for plotting later
+    """
+
+    fig, ax = plt.subplots(1, 1, figsize=(12, 5))
+
+    for color_ind, row in enumerate(sg_query):
+        color = dict(color=f"C{color_ind}")
+        wf = row["wf_similarity"]
+        ca = row["xcorrel_asymm"]
+        ax.scatter(wf, ca, **color)
+        ax.text(wf, ca, f"({row['unit1']},{row['unit2']})", **color)
+
+    ax.set_xlabel("waveform similarity")
+    ax.set_ylabel("cross-correlogram asymmetry")
+
+    plt.close()
+
+    return fig
+
+
+def calculate_ca(bins: np.ndarray, correl: np.ndarray) -> float:
+    """Calculate Correlogram Asymmetry (CA)
+
+    defined as the contrast ratio of the area of the correlogram right and
+    left of coincident activity (zero).
+    http://www.psy.vanderbilt.edu/faculty/roeaw/edgeinduction/Fig-W6.htm
+
+    Parameters
+    ----------
+    bins : np.ndarray
+        array of bin edges
+    correl : np.ndarray
+        array of correlogram values
+    """
+    if not len(bins) == len(correl):
+        raise ValueError("Mismatch in lengths for correl asymmetry")
+    right = np.sum(correl[bins > 0])
+    left = np.sum(correl[bins < 0])
+    return 0 if (right + left) == 0 else (right - left) / (right + left)
+
+
+def calculate_isi_violation(
+    peak1: np.ndarray, peak2: np.ndarray, isi_threshold_s: float = 1.5
+) -> float:
+    """Calculate isi violation between two spike trains"""
```

Comment:
numpy style docstring with expected shape of arrays would be good here.

Reply:
I think the peaks are derived from the calculated spike times. How would you describe that shape in a docstring?

Reply:
shape (n_spikes,)